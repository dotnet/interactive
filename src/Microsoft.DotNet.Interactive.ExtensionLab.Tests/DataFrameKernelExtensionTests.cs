// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System;
using System.Diagnostics;
using System.IO;
using System.Threading.Tasks;
using Assent;
using Microsoft.Data.Analysis;
using Microsoft.DotNet.Interactive.Formatting;
using Xunit;
using Xunit.Abstractions;

namespace Microsoft.DotNet.Interactive.ExtensionLab.Tests
{
    public class DataFrameKernelExtensionTests : IDisposable
    {

        private readonly ITestOutputHelper _output;
        private readonly Configuration _configuration;

        public DataFrameKernelExtensionTests(ITestOutputHelper output)
        {
            _output = output;
            _configuration = new Configuration()
                .SetInteractive(Debugger.IsAttached)
                .UsingExtension("json");
        }
        [Fact]
        public async Task it_registers_formatters()
        {
            using var kernel = new CompositeKernel();

            var kernelExtension = new DataFrameKernelExtension();

            await kernelExtension.OnLoadAsync(kernel);

            using var stream = new MemoryStream();
            using var writer = new StreamWriter(stream);
            writer.AutoFlush = true;
            writer.Write(@"id,name,color,deliciousness
1,apple,green,10
2,banana,yellow,11
3,cherry,red,9000");
            stream.Position = 0;

            var dataframe = DataFrame.LoadCsv(stream);

            var formatted = dataframe.ToDisplayString(TabularDataFormatter.MimeType);

            this.Assent(formatted, _configuration);
        }

        public void Dispose()
        {
            Formatter.ResetToDefault();
        }
    }
}