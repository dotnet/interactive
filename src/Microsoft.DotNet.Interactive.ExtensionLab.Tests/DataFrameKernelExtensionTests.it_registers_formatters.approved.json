{
  "schema": {
    "primaryKey": [],
    "fields": [
      {
        "name": "id",
        "type": "number"
      },
      {
        "name": "name",
        "type": "string"
      },
      {
        "name": "color",
        "type": "string"
      },
      {
        "name": "deliciousness",
        "type": "number"
      }
    ]
  },
  "data": [
    {
      "id": 1.0,
      "name": "apple",
      "color": "green",
      "deliciousness": 10.0
    },
    {
      "id": 2.0,
      "name": "banana",
      "color": "yellow",
      "deliciousness": 11.0
    },
    {
      "id": 3.0,
      "name": "cherry",
      "color": "red",
      "deliciousness": 9000.0
    }
  ]
}